web/PDFSinglePageViewer.js

```diff
@@ -0,0 +1,944 @@
+  }
+
+  getPageView(index) {
+    return this._pages[index];
+  }
+
+  /**
+   * @returns {boolean} true if all {PDFPageView} objects are initialized.
+   */
+  get pageViewsReady() {
+    return this._pageViewsReady;
+  }
+
+  /**
+   * @returns {number}
+   */
+  get currentPageNumber() {
+    return this._currentPageNumber;
+  }
+
+  /**
+   * @param {number} val - The page number.
+   */
+  set currentPageNumber(val) {
+    if ((val | 0) !== val) { // Ensure that `val` is an integer.
+      throw new Error('Invalid page number.');
+    }
+    if (!this.pdfDocument) {
+      return;
+    }
+    // The intent can be to just reset a scroll position and/or scale.
+    this._setCurrentPageNumber(val, /* resetCurrentPageView = */ true);
+  }
+
+  /**
+   * @private
+   */
+  _setCurrentPageNumber(val, resetCurrentPageView = false) {
+    if (this._currentPageNumber === val) {
+      if (resetCurrentPageView) {
+        this._resetCurrentPageView();
+      }
+      return;
+    }
+
+    if (!(0 < val && val <= this.pagesCount)) {
+      console.error(
+        `PDFSPViewer._setCurrentPageNumber: "${val}" is out of bounds.`);
+      return;
+    }
+
+    let arg = {
+      source: this,
+      pageNumber: val,
+      pageLabel: this._pageLabels && this._pageLabels[val - 1],
+    };
+    this._currentPageNumber = val;
+    this.eventBus.dispatch('pagechanging', arg);
+    this.eventBus.dispatch('pagechange', arg);
+
+    if (resetCurrentPageView) {
+      this._resetCurrentPageView();
+    }
+  }
+
+  /**
+   * @returns {string|null} Returns the current page label,
+   *                        or `null` if no page labels exist.
+   */
+  get currentPageLabel() {
+    return this._pageLabels && this._pageLabels[this._currentPageNumber - 1];
+  }
+
+  /**
+   * @param {string} val - The page label.
+   */
+  set currentPageLabel(val) {
+    let pageNumber = val | 0; // Fallback page number.
+    if (this._pageLabels) {
+      let i = this._pageLabels.indexOf(val);
+      if (i >= 0) {
+        pageNumber = i + 1;
+      }
+    }
+    this.currentPageNumber = pageNumber;
+  }
+
+  /**
+   * @returns {number}
+   */
+  get currentScale() {
+    return this._currentScale !== UNKNOWN_SCALE ? this._currentScale :
+                                                  DEFAULT_SCALE;
+  }
+
+  /**
+   * @param {number} val - Scale of the pages in percents.
+   */
+  set currentScale(val) {
+    if (isNaN(val)) {
+      throw new Error('Invalid numeric scale');
+    }
+    if (!this.pdfDocument) {
+      return;
+    }
+    this._setScale(val, false);
+  }
+
+  /**
+   * @returns {string}
+   */
+  get currentScaleValue() {
+    return this._currentScaleValue;
+  }
+
+  /**
+   * @param val - The scale of the pages (in percent or predefined value).
+   */
+  set currentScaleValue(val) {
+    if (!this.pdfDocument) {
+      return;
+    }
+    this._setScale(val, false);
+  }
+
+  /**
+   * @returns {number}
+   */
+  get pagesRotation() {
```

Comment:
Want is a strong word. I'm not very associated with the project.

And I definitely don't know about still wrt underscore/camel case